### PR TITLE
Feature/docs mam fixes

### LINKED ISSUE #2149: The multi-agent manager has a couple of bugs
**Describe the bug**
The multi-agent manager has a couple of bugs:
- `.build` folder and `soef_token.txt.` are created in the directory from which the MAM is used, not in the agent's directory
- the config overrides are not used during the `issue_certificates` run. Suggesting that the config overrides are not properly applied.

**To Reproduce**
Run MAM demo in docs and related test. Inspect output.

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes

fixes for issue certs and build dir for AEABuilder

## Fixes

#2149

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: aea/aea_builder.py [hunk 1]
```diff
@@ -939,9 +938,7 @@ def run_build_for_component_configuration(
             logger.info(f"Building package {config.component_id}...")
 
         source_directory = cast(str, config.directory)
-        target_directory = os.path.abspath(
-            os.path.join(root_dir, config.build_directory)
-        )
+        target_directory = os.path.abspath(config.build_directory)
```
Comment (DavidMinarsch) [inline aea/aea_builder.py:941]: I think this might be incorrect

File: aea/cli/issue_certificates.py [hunk 2]
```diff
@@ -43,29 +42,30 @@
...
     for connection_id in agent_config_manager.agent_config.connections:
-        _process_connection(ctx, agent_config_manager, connection_id)
-
-    click.echo("All certificates have been issued.")
+        _process_connection(path, agent_config_manager, connection_id)
 
 
 def _process_certificate(
-    ctx: Context, cert_request: CertRequest, connection_id: PublicId
+    path: str,
```
Comment (DavidMinarsch) [inline aea/cli/issue_certificates.py:58]: more descriptive name for this path

File: aea/cli/utils/package_utils.py [hunk 3]
```diff
@@ -186,12 +186,12 @@ def try_get_item_target_path(
 
 
 def get_package_path(
-    ctx: Context, item_type: str, public_id: PublicId, is_vendor: bool = True
+    path: str, item_type: str, public_id: PublicId, is_vendor: bool = True
```
Comment (DavidMinarsch) [inline aea/cli/utils/package_utils.py:189]: The problem is, that's not just a path. Needs to be better named.

File: aea/aea_builder.py (hunk 1 above)
Comment (solarw) [reply aea/aea_builder.py:941]: that's correct, config_build directory set with respect to root_build_directory set in AEABuilder

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Pull request merged**